src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1320 @@
+    }
+
+    // ================================================================================================
+
+    public long getFeatureOrderNumber() { return featureOrderNumber; }
+
+    public String getChromosomeName() {
+        return chromosomeName;
+    }
+
+    public AnnotationSource getAnnotationSource() {
+        return annotationSource;
+    }
+
+    public FeatureType getFeatureType() {
+        return featureType;
+    }
+
+    public int getGenomicStartLocation() {
+        return genomicStartLocation;
+    }
+
+    public int getGenomicEndLocation() {
+        return genomicEndLocation;
+    }
+
+    public GenomicStrand getGenomicStrand() {
+        return genomicStrand;
+    }
+
+    public GenomicPhase getGenomicPhase() {
+        return genomicPhase;
+    }
+
+    public String getGeneId() {
+        return geneId;
+    }
+
+    public String getTranscriptId() {
+        return transcriptId;
+    }
+
+    public GeneTranscriptType getGeneType() {
+        return geneType;
+    }
+
+    public String getGeneName() {
+        return geneName;
+    }
+
+    public GeneTranscriptType getTranscriptType() {
+        return transcriptType;
+    }
+
+    public String getTranscriptName() {
+        return transcriptName;
+    }
+
+    public int getExonNumber() {
+        return exonNumber;
+    }
+
+    public String getExonId() {
+        return exonId;
+    }
+
+    public LocusLevel getLocusLevel() {
+        return locusLevel;
+    }
+
+    public ArrayList<OptionalField<?>> getOptionalFields() {
+        return optionalFields;
+    }
+
+    public String getAnonymousOptionalFields() {
+        return anonymousOptionalFields;
+    }
+
+    public OptionalField<?> getOptionalField(String key) {
+        for (OptionalField<?> optionalField : optionalFields) {
+            if ( optionalField.getName().equals(key) ) {
+                return optionalField;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Comparable interface implementation for {@link GencodeGtfFeature}.
+     *
+     * Order is determined by {@link #featureOrderNumber}
+     *
+     * @param other {@link GencodeGtfFeature} to which to compare
+     * @return -1 if this < other; 0 if this == other; 1 if this > other
+     */
+    @Override
+    public int compareTo(GencodeGtfFeature other) {
+        return (int)(featureOrderNumber - other.featureOrderNumber);
+    }
+
+    /**
+     * Checks if {@code other} is contained within this {@link GencodeGtfFeature}.
+     * Comparison is made using {@link #genomicStartLocation} and {@link #genomicEndLocation} (both ends inclusive).
+     * @param other {@link GencodeGtfFeature} of which to check the bounds.
+     * @return true if {@code other} is contained within the bounds of this {@link GencodeGtfFeature}, false otherwise.
+     */
+    public boolean contains(GencodeGtfFeature other) {
+        return (other.getStart() >= getStart()) && (other.getEnd() <= getEnd());
```

Comment:
Yeah - I think it makes sense to store location in a SimpleInterval and use this to call into SimpleInterval::contains